smart_open/s3.py

```diff
@@ -52,6 +52,26 @@
 _OUT_OF_RANGE = 'InvalidRange'
 
 
+class _ClientWrapper:
+    """Wraps a client to inject the appropriate keyword args into each method call.
+
+    The keyword args are a dictionary keyed by the fully qualified method name.
+    For example, S3.Client.create_multipart_upload.
+
+    See https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#client
```

Comment:
Deserve moar examples & explanations in README

Reply:
I added more info to the howto and help.txt - it's a bit too low-level for the README.